sdk/eventhub/azure-eventhubs/azure/eventhub/aio/eventprocessor/partition_context.py

```diff
@@ -12,28 +12,26 @@ class PartitionContext(object):
 
     Users can use update_checkpoint() of this class to save checkpoint data.
     """
-    def __init__(self, eventhub_name: str, consumer_group_name: str,
-                 partition_id: str, owner_id: str, partition_manager: PartitionManager):
+    def __init__(self, fully_qualified_namespace: str, eventhub_name: str, consumer_group_name: str,
+                 partition_id: str, owner_id: str, partition_manager: PartitionManager = None):
+        self.fully_qualified_namespace = fully_qualified_namespace
         self.partition_id = partition_id
         self.eventhub_name = eventhub_name
         self.consumer_group_name = consumer_group_name
         self.owner_id = owner_id
         self._partition_manager = partition_manager
 
-    async def update_checkpoint(self, offset, sequence_number=None):
+    async def update_checkpoint(self, event):
         """
         Updates the checkpoint using the given information for the associated partition and consumer group in the
         chosen storage service.
 
-        :param offset: The offset of the ~azure.eventhub.EventData the new checkpoint will be associated with.
-        :type offset: str
-        :param sequence_number: The sequence_number of the ~azure.eventhub.EventData the new checkpoint will be
-         associated with.
-        :type sequence_number: int
+        :param event: The EventData instance. Its offset and sequence number are to be saved in the checkpoint store.
+        :type event: EventData
         :return: None
         """
-        # TODO: whether change this method to accept event_data as well
-        await self._partition_manager.update_checkpoint(
-            self.eventhub_name, self.consumer_group_name, self.partition_id, self.owner_id, offset,
-            sequence_number
-        )
+        if self._partition_manager:
```

Comment:
So if not partition manager we just do nothing? No error, no warning, just nothing?

Reply:
Added logging for else
```python
_LOGGER.info(
                "namespace %r, eventhub %r, consumer_group %r, partition_id %r "
                "update_checkpoint is called without partition manager. No checkpoint is updated.",
                self.fully_qualified_namespace, self.eventhub_name, self.consumer_group_name, self.partition_id)
```